### ISSUE TITLE
BeginDragDropSource() only working with mouse left button

### ISSUE THREAD

**Issue by e3dos** [ocornut/imgui#1637]
I tried other mouse buttons, but only 0 on the second parameter of BeginDragDropSource() will work.

```cpp
(ImGui::BeginDragDropSource(0,2); //doesn't work, dragging doesn't start
```
I'm on sdl2 and opengl

**Comment by ocornut** [comment]
Hmm, ok this is a problem. The issue is that by default `BeginDragDropSource()` relies on state of whether the last item is active, which is most typically set by the button behavior handling the left-button. 

Could you describe your use case so I can think of a solution knowing the context?

**Comment by exsec** [comment]
What I'm doing is a drag and drop to reorder a treelist.
the problem is I already use the left mouse button to process the click and select the node. So if I try to drag multiple selected tree nodes, the left mouse click is processed before the drag and I end up with only one selection.
tldr I'm trying to drag multiple selections.
That's why I wanted to use another mouse button for the drag, middle mouse button in my case.

**Comment by ocornut** [comment]
> the problem is I already use the left mouse button to process the click and select the node. So if I try to drag multiple selected tree nodes, the left mouse click is processed before the drag and I end up with only one selection.

How are you handling the selection?
Selection should happen on click+release not on click, that's fairly common behavior and then you don't have this problem. The "Selectable Multiple Selection" example has those and they would work with Left-Button Drag & Drop afaik.

**Comment by ocornut** [comment]
FYI I am going to remove the `mouse_button = 0` parameter (right now) until this is fixed.

**Comment by exsec** [comment]
Ok, I'm using ImGui::IsItemClicked()
I'll try your suggestion and see how it works, thanks.

**Comment by ngminteck** [comment]
same issue as i did will try mouse click + mouse release

https://github.com/ocornut/imgui/issues/2209

**Comment by ngminteck** [comment]
click+release don't work for me

**Comment by ocornut** [comment]
As per #3885, pushed 954b06af

This works:
```cpp
bool open = ImGui::TreeNode("TreeNode");

ImGuiWindow* window = ImGui::GetCurrentWindow();
ImGui::ButtonBehavior(window->DC.LastItemRect, window->DC.LastItemId, NULL, NULL, ImGuiButtonFlags_MouseButtonMiddle | ImGuiButtonFlags_MouseButtonRight);

if (ImGui::BeginDragDropSource())
{
    ImGui::Text("Dragged");
    ImGui::EndDragDropSource();
}
if (open)
    ImGui::TreePop();
```

There are two sides to it:
- unusual use of ButtonBehavior to retrofit multi-button checks in an existing which (which could be also handled by adding new flags to TreeNode, etc.).
- added support in BeginDragDropSource()
-
